src/annotator/anchoring/pdf.js

```diff
@@ -269,6 +256,29 @@ async function findPageByOffset(offset) {
   return { index: viewer.pagesCount - 1, offset: pageStartOffset, text };
 }
 
+/**
+ * Return true if `char` is an ASCII space.
+ *
+ * This is more efficient than `/\s/.test(char)` but does not handle Unicode
+ * spaces.
```

Comment:
> Does this imply that encoding in text layers is ASCII-only (surely not?) or that we don't consider Unicode spaces to be whitespace (more likely?)? 

Your second try is right, we just treat non-ASCII spaces as not being spaces. That's acceptable here because we are trying to filter out the whitespace changes from one PDF.js version to another and, AFAIK, those differences are always in ASCII space chars.

> What's the efficiency difference here? 

According to some [ESBench tests](https://esbench.com/bench/624bf0486c89f600a57016f6), about 2-3x depending on the browser.